src/pudl/extract/excel.py

```diff
@@ -0,0 +1,254 @@
+    into dataframes, metadata tells us how to do this. Metadata generally informs
+    us about the position of a given page in the file (which sheet and which row)
+    and it informs us how to translate excel column names into standardized
+    column names.
+
+    When metadata object is instantiated, it is given ${dataset} name and it
+    will attempt to load csv files from pudl.package_data.meta.xlsx_maps.${dataset}
+    package.
+
+    It expects the following kinds of files:
+    - skiprows.csv tells us how many initial rows should be skipped when loading
+      data for given (year, page).
+    - tab_map.csv tells us what is the excel sheet name that should be read
+      when loading data for given (year, page)
+    - coumn_map/${page}.csv currently informs us how to translate input column
+      names to standardized pudl names for given (year, input_col_name). Relevant
+      page is encoded in the filename.
+    """
+
+    def __init__(self, dataset_name):
+        """Create Metadata object and load metadata from python package.
+
+        Args:
+          dataset_name: Name of the package/dataset to load the metadata from.
+            Files will be loaded from pudl.package_data.meta.xlsx_meta.${dataset_name}.
+        """
+        pkg = f'pudl.package_data.meta.xlsx_maps.{dataset_name}'
+        self._dataset_name = dataset_name
+        self._skiprows = self._load_csv(pkg, 'skiprows.csv')
+        self._sheet_name = self._load_csv(pkg, 'tab_map.csv')
+        column_map_pkg = pkg + '.column_maps'
+        self._column_map = {}
+        for res in importlib.resources.contents(column_map_pkg):
+            # res is expected to be ${page}.csv
+            parts = res.split('.')
+            if len(parts) != 2 or parts[1] != 'csv':
+                continue
+            column_map = self._load_csv(column_map_pkg, res)
+            self._column_map[parts[0]] = column_map
+
+    def get_dataset_name(self):
+        return self._dataset_name
+
+    def get_sheet_name(self, year, page):
+        """Returns name of the excel sheet that contains the data for given year and page."""
+        return self._sheet_name.at[year, page]
+
+    def get_skiprows(self, year, page):
+        """Returns number of initial rows to skip when loading given year and page."""
+        return self._skiprows.at[year, page]
+
+    def get_column_map(self, year, page):
+        """Returns the dictionary mapping input columns to pudl columns for given year and page."""
+        return {v: k for k, v in self._column_map[page].loc[year].to_dict().items()}
+
+    def get_all_columns(self, page):
+        """Returns set of all pudl (standardized) columns for a given page (across all years)."""
+        return set(self._column_map[page].columns)
+
+    def get_all_pages(self):
+        """Returns list of all known pages."""
+        return self._column_map.keys()
+
+    def _load_csv(self, package, filename):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/zaneselvans/pudl/pullRequest?prid=5191218)